main.go

```diff
@@ -71,7 +79,7 @@ func main() {
 	clients.Healthcheck = &healthcheck
 
 	if err = registerCheckers(ctx, clients); err != nil {
-		os.Exit(1)
+		return err
 	}
 
```

Comment:
Line 100 - Should return error

Reply:
Shall I create an error channel and log it?

Reply:
But it's a separate go routine. I think in some services we had a channel, and then if it was closed, we would start the graceful shutdown - but after the requirement to not kill services if dependencies fail, the channel didn't have any propose (we log anyway from the go routine if an error happens)

Reply:
I think we should be forcing a graceful shutdown here as this application has only one responsibility and that is to handle requests. If the server is an issue then the application can't function. I would force a graceful shutdown here